sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -207,15 +310,52 @@ private[spark] class HiveExternalCatalog(client: HiveClient, hadoopConf: Configu
     assert(tableDefinition.identifier.database.isDefined)
     val db = tableDefinition.identifier.database.get
     requireTableExists(db, tableDefinition.identifier.table)
+    verifyTableProperties(tableDefinition)
```

Comment:
alterTable is my main concern. It is possible that `tableDefinition` is in the standard form. But, the actual metadata stored in the metastore is using properties to store schema. Then, the alterTable may just fail or significantly change the metadata in metastore.